File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.cc
#include <bits/stdc++.h>
using namespace std;

int n, a[1002], cnt[1002], ret;

int main(){
	
	cin >> n;
	for(int i=0; i<n; i++){
		scanf("%d", &a[i]);
	}
	
	for(int i=0; i<n; i++){
		int maxValue = 0;
		for(int j=0; j<i; j++){
			if(a[j]<a[i] && maxValue < cnt[j]){
				maxValue = cnt[j];
			}
		}
		cnt[i] = maxValue+1;
		ret = max(ret, cnt[i]);
	}
	cout << ret << '\n';
	
	return 0;
}

File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n] = input.shift().split(" ").map(Number);
const a = input[0].split(" ").map(Number);
const cnt = new Array(1002);
let ret=1;

for(let i=0; i<n; i++){
  let maxValue = 0;
  for(let j=0; j<i; j++){
    if(a[j]<a[i] && maxValue<cnt[j]){
      maxValue = cnt[j];
    }
  }
  cnt[i] = maxValue + 1;
  ret = Math.max(ret, cnt[i]);
}

console.log(ret);